src/request.rs

```diff
@@ -887,6 +885,47 @@ impl<'a> Request<'a> {
     ///let response = Request::new(&uri).send(&mut writer).unwrap();
     ///```
     pub fn send<T: Write>(&self, writer: &mut T) -> Result<Response, error::Error> {
+        if self.inner.uri.scheme() == "https" {
+            let mut cnf = tls::Config::default();
+            let cnf = match self.root_cert_file_pem {
+                Some(p) => cnf.add_root_cert_file_pem(p)?,
+                None => &mut cnf,
+            };
+            self.send_with_config(writer, Some(cnf))
+        } else {
+            self.send_with_config(writer, None)
+        }
+    }
+
+    /// Sends a HTTPs request. The connection will only be established if the supplied certificate
+    /// matches the server's root certificate.
+    /// Supply the content of the server's root certificate as a String. There are no predefined trusted
+    /// root certificates in the tls config.
+    pub fn send_with_pem_certificate<T: Write>(
+        &self,
+        writer: &mut T,
+        certificate_content: Option<String>,
+    ) -> Result<Response, error::Error> {
+        if self.inner.uri.scheme() == "https" {
+            let mut cnf = tls::Config::empty_root_store();
+            let cnf = match certificate_content {
+                Some(c) => cnf.add_root_cert_content_pem_file(&c)?,
+                None => &mut cnf,
+            };
+            self.send_with_config(writer, Some(cnf))
+        } else {
+            println!("Send with certificate, but not a https request!");
```

Comment:
Did this crate already have `println!` before your changes? A library IMO should not use `println!` but rather put the information into the error, or the log otherwise.

Reply:
They have println! but I will add the logs

Reply:
If you have some time, that would be nice, yes